drools/src/main/java/com/datatorrent/drools/operator/StatelessDroolsOperator.java

```diff
@@ -43,22 +44,74 @@
 public class StatelessDroolsOperator extends BaseOperator implements ActivationListener<Context.OperatorContext>
 {
   private static final Logger LOG = LoggerFactory.getLogger(StatelessDroolsOperator.class);
+  private static final int DEFAULT_BATCH_SIZE = 1000;
+  public final transient DefaultOutputPort<Object> factsOutput = new DefaultOutputPort<>();
+  public final transient DefaultOutputPort<Map<Rule, Integer>> ruleCountOutput = new DefaultOutputPort<>();
+  public final transient DefaultOutputPort<Map<Object, List<Rule>>> factAndFiredRulesOutput = new DefaultOutputPort<>();
   private transient StatelessKieSession kieSession;
   private String rulesDir;
   private boolean loadSpringSession = false;
   private String sessionName;
   private Map<Object, List<Rule>> factsAndFiredRules;
   private Map<Rule, Integer> ruleCount;
+  private List<Object> facts;
+  private List<Object> factsFromRules;
+  private int batchSize = DEFAULT_BATCH_SIZE;
   public final transient DefaultInputPort<Object> factsInput = new DefaultInputPort<Object>()
   {
     @Override
     public void process(Object tuple)
     {
-
+      facts.add(tuple);
+      if (facts.size() >= batchSize) {
+        executeRules();
+      }
     }
   };
 
   @Override
+  public void endWindow()
+  {
+    super.endWindow();
+    if (facts.size() > 0) {
+      executeRules();
+    }
+    factAndFiredRulesOutput.emit(factsAndFiredRules);
```

Comment:
I hope you mean the entries of the map. It doesn't makes sense to do it one by one. The ordering of the tuples is anyways not guranteed